setup.py

```diff
@@ -118,7 +118,7 @@
         "hexbytes>=0.3.1,<2",
         "py-geth>=5.4.0,<6",
         "trie>=3.0.1,<4",  # Peer: stricter pin needed for uv support.
-        "web3[tester]>=6.20.1,<8",
+        "web3[tester]>=7.11,<8",
```

Comment:
By doing this, we can get rid of the all the web3 compat imports we have to do. That may be worth doing on a separate PR

Reply:
Actually I think I can get around this via `try..except` on `eth_account.Account.sign_authorization` below (which will raise `AttributeError` on un-supported web3py versions)